ludwig/callbacks.py

```diff
@@ -116,16 +116,16 @@ def on_test_start(self, trainer, progress_tracker, save_path):
     def on_test_end(self, trainer, progress_tracker, save_path):
         pass
 
-    def on_build_metadata_start(self, df):
+    def on_build_metadata_start(self, df, dataset_name):
```

Comment:
`dataset_name` should be consistent with the other param if possible, so maybe we can change this to `mode`.

Reply:
Done.